Automattic/wp-calypso
# Tiled Gallery: Introduce image srcset

Add a `srcset` with a reasonable set of photonized image sizes so that the browser can pick an optimized image.

The image sizes are conservative because we know very little about themes or how the image may be used when we generate the srcset.

This is a fairly naïve implementation, if we push the srcset generation deeper where we know images will be positioned, we could further optimize. That has the potential to add endless complexity 🙂 

A deprecated version of the block is added to handle migration to the newer output with srcsets

Resolves #29787
Fixes #30118

Related conversation about allowing some attributes: p3btAN-12W-p2

## Known issues
- WordPress.com sites appear to strip the `srcset` attribute for some reason. This means that the blocks are always invalidated. D24378-code includes a fix (depends on)

## To do
- [x] Save (serialized) srcset
- [x] Migrate (handle invalid deprecated blocks)
- [x] Show some loading image UI (#30758)
- [x] Confirm opening previous version of the block works
- [ ] Remove deprecate and `save` srcset (handled in PHP render by https://github.com/Automattic/jetpack/pull/11397)

## Testing

gutenpack-jn

- Add a tiled gallery using current master and save.
- Open the same post with this branch. You should see an "invalid block" warning in the console, but the deprecation handles it.
- Save the post.
- Reload ❗️ WordPress.com sites will invalidate the block here ❗️ 
- View the gallery on the frontend. Observe the srcset in action

To observe the srcset in action, start with a narrow viewport and observe the image requests (filter img requests `/(i[0-2]\.wp\.com)|(files\.wordpress\.com)/` in `network`).

Increase the browser size and notice that larger assets are requested.
